xarray/core/dataset.py

```diff
@@ -3393,11 +3399,20 @@ def roll(self, **shifts):
         if invalid:
             raise ValueError("dimensions %r do not exist" % invalid)
 
+        if roll_coords is None:
+            warnings.warn("roll_coords will be set to False in the future."
+                          " Explicitly set roll_coords to silence warning.",
+                          DeprecationWarning, stacklevel=3)
```

Comment:
Use `FutureWarning` here, which doesn't get silenced by default and is intended for warning about future changes in behavior.